experimental/HttpBenchmark/Program.cs

```diff
@@ -0,0 +1,375 @@
+                    {
+                        duration = int.Parse(durationOption.Value());
+                    }
+
+                    var pipeline = _defaultPipeline;
+                    if (pipelineOption.HasValue())
+                    {
+                        pipeline = int.Parse(pipelineOption.Value());
+                    }
+
+                    var keepalive = _defaultKeepalive;
+                    if (keepaliveOption.HasValue())
+                    {
+                        keepalive = bool.Parse(keepaliveOption.Value());
+                    }
+
+                    var method = _defaultMethod;
+                    if (methodOption.HasValue())
+                    {
+                        method = methodOption.Value();
+                    }
+                    method = method.ToUpperInvariant();
+
+                    var body = bodyOption.Value();
+
+                    Uri uri = null;
+                    if (!string.IsNullOrWhiteSpace(urlArgument.Value))
+                    {
+                        uri = new Uri(urlArgument.Value);
+                    }
+
+                    if (connections <= 0 ||
+                        duration <= 0 ||
+                        pipeline <= 0 ||
+                        uri == null)
+                    {
+                        app.ShowHelp();
+                        return 2;
+                    }
+
+                    return Run(connections, TimeSpan.FromSeconds(duration), pipeline, keepalive, method, body, uri);
+                }
+                catch (Exception e)
+                {
+                    app.ShowHelp();
+                    Console.WriteLine(e);
+                    return 2;
+                }
+            });
+
+            return app.Execute(args);
+        }
+
+        private static int Run(int connections, TimeSpan duration, int pipeline, bool keepalive, string method, string body, Uri uri)
+        {
+            Init(connections, duration, pipeline, keepalive, uri);
+
+            var requestString =
+                $"{method} {uri.PathAndQuery} HTTP/1.1\r\n" +
+                $"Host: {uri.Host}:{uri.Port}\r\n" +
+                $"Accept: */*\r\n" +
+                $"\r\n" +
+                body;
+
+            if (!string.IsNullOrEmpty(body))
+            {
+                requestString = requestString.Insert(requestString.IndexOf("Accept: */*\r\n"),
+                    $"Content-Length: {Encoding.ASCII.GetBytes(body).Length}\r\n");
+            }
+
+            var requestBytes = Encoding.ASCII.GetBytes(requestString);
+
+            var responseLength = -1;
+
+            // Calculate response length
+            using (var socket = CreateSocket(uri).Result)
+            {
+                Console.WriteLine("Request:");
+                Console.Write(requestString);
+                socket.Send(requestBytes);
+
+                var responseBuffer = new byte[1024];
+                int bytesReceived = 0;
```

Comment:
I updated the tool to support responses up to 1 MB, rather than the 1KB previously supported.  The `bytesReceived` is still fine as an `int`, unless we want to support responses larger than `int.MaxValue`.